### ISSUE TITLE
Publishing is not incremental

### ISSUE THREAD

**Issue by tmat** [dotnet/sdk#2573]
Repro project:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <Target Name="PublishThisProject" AfterTargets="Build" DependsOnTargets="Publish"/>
</Project>
```

Run msbuild twice:
`msbuild /bl:Repro1.binlog`
`msbuild /bl:Repro2.binlog`

The files are copied twice. 

![image](https://user-images.githubusercontent.com/41759/46638412-9f2bce00-cb15-11e8-9104-7866f9a2ebe4.png)

**Comment by tmat** [comment]
@nguerrera FYI

**Comment by livarcocc** [comment]
@peterhuene

**Comment by peterhuene** [comment]
I'll see what I can do to fix this as part of the copy-local for build changes.

**Comment by peterhuene** [comment]
@tmat this has been fixed in `release/3.0.1xx` and will be in the next preview release.  Thanks!
